scipy/sparse/tests/test_base.py

```diff
@@ -2347,15 +2367,21 @@ def test_inplace_success(self):
         bp = bp + a
         assert_allclose(b.toarray(), bp.toarray())
 
-        b *= a
-        bp = bp * a
+        if isinstance(b, sparray):
+            b *= a
+            bp = bp * a
+            assert_allclose(b.toarray(), bp.toarray())
+
+        b @= a
```

Comment:
Wouldn't this have the same issue with `__imatmul__` support that we try/except around above?

Reply:
This `__imatmul__` is for sparse @ sparse which is supported.  
We used try/catch above when `imatmul`ing a dense `np.array` by a sparse. `np.array` does not implement imatmul with sparse so it is not  which is not supported.